### PR TITLE
Set Wallet Status to 'Ready' after correctly authenticating wallet

### PR DESCRIPTION
**Pull request by zfox23**
The big thing here is that I'm now forcing the wallet status to be "READY" as soon as the user properly authenticates the wallet with their passphrase. This should have been done from the beginning, instead of us relying on some other method checking and updating the status in another location.

This fixes [Manuscript #10485](https://highfidelity.manuscript.com/f/cases/10485/Marketplace-Your-Wallet-is-not-set-up-notification-appears-after-rezzing-a-newly-purchased-free-item) (and probably some other bugs).

**Test Plan:**
1. Start up Interface logged in, but don't authenticate your wallet
2. Open up marketplace and search for "Jimi the snail"
3. Buy a copy of "Jimi the snail". Enter your wallet passphrase when prompted
4. On the confirmation page after purchasing, rez the item
5. Verify that you DO NOT see a notification in the top right of Interface that your "wallet isn't set up"
    - Unfortunately, due to another (unrelated) bug, Jimi the snail won't pass static certificate verification, so he won't rez. That's not what this PR addresses.

### CODE REVIEW

**Comment by Nex-Pro** [comment]
Will test when win build is available. @zfox23

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5599/HighFidelity-Beta-PR11972-623e6395be15677d2394587489e36ed58020de17.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5599/HighFidelity-Beta-PR11972-623e6395be15677d2394587489e36ed58020de17.exe

**Comment by Nex-Pro** [comment]
Testing Win10

**Comment by Nex-Pro** [comment]
### PASS
Works as per notes.

Side note:
Have informed @zfox23 that user can click on "My Purchases" before having entered wallet passphrase.

File: interface/src/scripting/WalletScriptingInterface.h [hunk 1]
```diff
@@ -39,7 +39,7 @@ class WalletScriptingInterface : public QObject, public Dependency {
 
     Q_INVOKABLE void refreshWalletStatus();
     Q_INVOKABLE uint getWalletStatus() { return _walletStatus; }
-    void setWalletStatus(const uint& status) { _walletStatus = status; }
+    void setWalletStatus(const uint& status);
```
Comment (howard-stearns) [inline interface/src/scripting/WalletScriptingInterface.h]: Let's add a comment that this should not be made invokable such that scripts could lie to you.

Comment (zfox23) [reply interface/src/scripting/WalletScriptingInterface.h]: Will do!

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5612/HighFidelity-Beta-PR11972-efd5c0ccb7d1d4a40d88784bfa6c36b3b8e8edc5.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5612/HighFidelity-Beta-PR11972-efd5c0ccb7d1d4a40d88784bfa6c36b3b8e8edc5.dmg

**Pull request merged**